Tools/scripts/smelly.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python
+# Script checking if all symbols exported by libpython start with "Py" or "_Py"
+
+import subprocess
+import sys
+import sysconfig
+
+
+def get_exported_symbols():
+    LIBRARY = sysconfig.get_config_var('LIBRARY')
+    if not LIBRARY:
+        raise Exception("failed to get LIBRARY")
+
+    args = ('nm', '-p', LIBRARY)
+    print("+ %s" % ' '.join(args))
+    proc = subprocess.run(args, stdout=subprocess.PIPE, universal_newlines=True)
+    if proc.returncode:
+        sys.stdout.write(proc.stdout)
+        sys.exit(proc.returncode)
+
+    stdout = proc.stdout.rstrip()
+    if not stdout:
+        raise Exception("command output is empty")
+    return stdout
+
+
+def get_smelly_symbols(stdout):
+    symbols = []
+    ignored_symtypes = set()
+    for line in stdout.splitlines():
+        # Split line '0000000000001b80 D PyTextIOWrapper_Type'
+        if not line:
+            continue
+
+        parts = line.split(maxsplit=2)
+        if len(parts) < 3:
+            continue
+
+        symtype = parts[1].strip()
+        # Ignore private symbols.
+        #
+        # If lowercase, the symbol is usually local; if uppercase, the symbol
+        # is global (external).  There are however a few lowercase symbols that
+        # are shown for special global symbols ("u", "v" and "w").
+        if symtype.islower() and symtype not in ("u", "v" "w"):
```

Comment:
Is the lack of comma between 'v' and 'w' intended or is it just a typo ?

Reply:
Oops, it's a bug. Fixed.